frontend/src/app/components/MainInstallButton.js

```diff
@@ -88,16 +102,22 @@ export default class MainInstallButton extends React.Component {
     if (isFirefox && isMobile) {
       return (
         <div>
-          <span data-l10n-id="landingRequiresDesktop">Test Pilot requires Firefox for Desktop on Windows, Mac or Linux</span>
+          <Localized id="landingRequiresDesktop">
+            <span>Test Pilot requires Firefox for Desktop on Windows, Mac or Linux</span>
+          </Localized>
         </div>
       );
     }
     return (
       <div>
           {!isFirefox ? (
-              <span data-l10n-id="landingDownloadFirefoxDesc" className="parens">(Test Pilot is available for Firefox on Windows, OS X and Linux)</span>
+              <Localized id="landingDownloadFirefoxDesc">
+                <span className="parens">(Test Pilot is available for Firefox on Windows, OS X and Linux)</span>
+              </Localized>
             ) : (
-              <span className="parens" data-l10n-id="landingUpgradeDesc2" data-l10n-args={JSON.stringify({ version: config.minFirefoxVersion })}>Test Pilot requires Firefox { config.minFirefoxVersion } or higher.</span>
+              <Localized id="landingUpgradeDesc2" $version={config.minFirefoxVersion}>
+                <span className="parens">Test Pilot requires Firefox { config.minFirefoxVersion } or higher.</span>
```

Comment:
I don't have much of an opinion on this. My gut feeling is to go with the valid jsx. Even though the nodes will just be tossed, it doesn't seem like that big of an issue.